### PR TITLE
Numpy frontend item

### PR DESCRIPTION
**Pull request by marschattha**
#15453

### CODE REVIEW

**Comment by marschattha** [comment]
@CatB1t could you review when feasible?
Not sure why some of these tests are failing.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 1]
```diff
@@ -1635,7 +1635,7 @@ def dtype_array_index(
         index_type = draw(st.sampled_from(["int", "ellipsis", "slice"]))
         if not allow_slices or index_type == "int":
             index += (draw(st.integers(min_value=-s + 1, max_value=s - 1)),)
-        if index_type == "ellipsis" and Ellipsis not in index:
+        elif index_type == "ellipsis" and Ellipsis not in index:
```
[the remark this answers is not in the document]
Comment (marschattha) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:1638]: If we don't change it to `elif` and leave it with `if`, the `allow_slices` flag/param does not actually work. 
And we end up with non integer indexes (slices) as well in the return index.

**Comment by marschattha** [comment]
@CatB1t @Aarsh2001 any updates?

**Review by Aarsh2001 — APPROVED** [review]
LGTM ! @CatB1t can you have a look at the changes in the helper files once, thanks 🙂

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by marschattha** [comment]
@CatB1t

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by Aarsh2001 — APPROVED** [review]
LGTM

**Comment by Aarsh2001** [comment]
Can you please fix the merge conflicts ? We'll then merge this PR ASAP, thanks 🙂

**Comment by marschattha** [comment]
> Can you please fix the merge conflicts ? We'll then merge this PR ASAP, thanks 🙂

Had to do more than fix the conflicts, but should be good now.

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 2]
```diff
@@ -1851,6 +1851,7 @@ def dtype_array_query(
             max_size=len(shape),
         )
     )
+
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:1854]: Hello @marschattha, can you kindly please revert this line here? this update seems irrelevant to the PR. Thanks in advance!

**Review by CatB1t — CHANGES_REQUESTED** [review]

**Comment by CatB1t** [comment]
Thanks @marschattha, @Aarsh2001 can take the lead on the review from here!

**Review by Aarsh2001 — APPROVED** [review]
My apologies for the delay, LGTM !

**Pull request merged**